test/core/end2end/h2_ssl_cert_test.cc

```diff
@@ -272,25 +270,11 @@ static void drain_cq(grpc_completion_queue* cq) {
 
 static void shutdown_server(grpc_end2end_test_fixture* f) {
   if (!f->server) return;
-  /* Perform a completion queue next, so that any pending operations can be
-   * finished, and resources can be released. This is so that, shutdown does not
-   * hang. For example, the server might be stuck in the handshaking code, which
-   * keeps a ref to a listener. Unless, it is unref'd, shutdown won't be able
-   * to proceed.
-   *
-   * (If shutdown times out, it is probably because 100ms wasn't enough. In that
-   * case, the deadline can be increased. Or, we could simply have another
-   * thread for the server to poll the completion queue while the shutdown
-   * progresses.)
-   */
-  GPR_ASSERT(grpc_completion_queue_next(
-                 f->cq, grpc_timeout_milliseconds_to_deadline(100), nullptr)
-                 .type == GRPC_QUEUE_TIMEOUT);
-  grpc_server_shutdown_and_notify(f->server, f->shutdown_cq, tag(1000));
-  GPR_ASSERT(grpc_completion_queue_pluck(f->shutdown_cq, tag(1000),
-                                         grpc_timeout_seconds_to_deadline(5),
-                                         nullptr)
-                 .type == GRPC_OP_COMPLETE);
+  grpc_server_shutdown_and_notify(f->server, f->cq, tag(1000));
+  grpc_event ev = grpc_completion_queue_next(
+      f->cq, grpc_timeout_seconds_to_deadline(5), nullptr);
+  ASSERT_EQ(ev.type, GRPC_OP_COMPLETE);
+  ASSERT_EQ(ev.tag, tag(1000));
```

Comment:
While this should be ideally true, I am worried that if the `f->cq` has some other pending operation (i.e with a different tag), then `grpc_completion_queue_next` may return that tag instead.
(**Note**: This ideally shouldn't happen but see that in `end_test(...)` function, we actually call `drain_cq(f->cq)` which just keeps calling completion queue next in a loop to drain all events.)

I suggest doing the following instead:
Call  the following here:
``` C++
 grpc_server_shutdown_and_notify(f->server, f->cq, tag(1000));
 grpc_completion_queue_shutdown(f->cq);
 drain_cq(f->cq); // This should only return once all the events (including the server shutdown) are done
```

And delete the following lines from the function `end_test(...)`
```C++
  grpc_completion_queue_shutdown(f->cq);
  drain_cq(f->cq)
```

Reply:
Done!